### PR TITLE
bt-638: company talent hire action

### LINKED ISSUE #638: Company talent hiring info
We need implement logic to hiring info.

Backend:
- [x] create relation table with companyId => userDetailsTable, talentId => userDetailsTable and other columns which we provide while hiring
- [x] create repository with create method and method to get all records about hiring
- [x] create service with such method
- [x] create endpoint GET and POST

Frontend:
- [x] implement hiring process (talent, company)
- [x] add list of hiring record in admin layout (Maybe add another page for this) - https://mui.com/material-ui/react-table/

Screenshots: 
Image

Image

**Please clarify hiring process with QA team !!!**

### PR DESCRIPTION
**Pull request by savinaDaria**
- Merged with #702 to get open page.
- Added hired radio-button handling in candidate open page
- Changed admin connections page to have only table with hirings

image
image

### CODE REVIEW

File: frontend/src/bundles/search-candidates/store/actions.ts [hunk 1]
```diff
@@ -26,13 +26,22 @@ const setFilters = createAsyncThunk<
...
+    },
     AsyncThunkConfig
 >(
     `${sliceName}/get-candidate-details`,
-    async (findPayload, { extra, rejectWithValue }) => {
+    async (findPayload, { extra, rejectWithValue, getState }) => {
+        const { searchCandidates } = getState();
         const { talentOnBoardingApi } = extra;
-
+        if (searchCandidates.filteredCandidates.length > 0) {
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/search-candidates/store/actions.ts]: Magic number

File: frontend/src/bundles/hiring-info/store/actions.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const getAllHiringInfo = createAsyncThunk<
+    HiringInfoFindResponseDto[],
+    undefined,
+    AsyncThunkConfig
+>(`${sliceName}/get-all-hiring-info`, async (_, { extra }) => {
+    const { hiringInfoApi } = extra;
+
+    const hiringInfo = await hiringInfoApi.getAllHiringInfo();
+
+    return hiringInfo.items;
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/hiring-info/store/actions.ts]:
```suggestion
    const { items } = await hiringInfoApi.getAllHiringInfo();

    return items;
```

**Review by MaksymGavatuk — APPROVED** [review]
Can be merged, just a few comments !

**Review by nikita-remeslov — APPROVED** [review]
Conflicts

**Pull request merged**